### ISSUE TITLE
Form Submission Fails ~~On Success~~

### ISSUE THREAD

**Issue by ishotjr** [FlipFlopHome/flipflop#2]
### Summary

Connecting to ssid appears to result in inability to return response to browser on `192.168.4.1`.
### Steps to Repro
1. enter correct ssid and password
2. submit
3. connection status page not displayed because `192.168.4.1` no longer accessible

![2015-07-18 12 37 35](https://cloud.githubusercontent.com/assets/1594827/8763345/46521bc8-2d65-11e5-86a5-77e711a13e78.png)
### Technical Details
- Build version: https://github.com/FlipFlopHome/flipflop/commit/bacb5c37e898abcd96d79cf87a334951fdad9f93
- ESP-12

**Comment by ishotjr** [comment]
@juliendarrah Looks like I may have been mistaken - the connection was not successful - perhaps related to #1 i.e. non-alpha characters in my password:

```
Toggle
Start Configuration AP Mode
AP Mode Connected
AP Mode Connected
Data Received
AP Mode Disconnected
Data Received
AP Mode Disconnected
AP Mode Connected
Data Received
AP Mode Disconnected
INVALID PASSWORD
 ets Jan  8 2013,rst cause:1, boot mode:(3,7)

load 0x40100000, len 30768, room 16
tail 0
chksum 0x5a
load 0x3ffe8000, len 1332, room 8
tail 12
chksum 0x66
ho 0 tail 12 room 4
load 0x3ffe8540, len 7804, room 12
tail 0
chksum 0xd7
csum 0xd7
rl▒
```

**Comment by juliendarrah** [comment]
This issue is somewhat related to issue-1.
I tested with an invalid SSID (containing non alphanumeric characters) and got a response page with invalid SSID name.
I then used a valid SSID with an invalid password and never got the response page back.
Need to check if we need a longer timeout. Right now it is set to 10sec.

**Comment by ishotjr** [comment]
@juliendarrah Agreed; my case was the latter: pure alpha SSID but special chars in the password (to be clear, password was _correct_ (I tried twice, pasting the second time) - only "invalid" in terms of the (erroneous) limitations of the system! :smirk: ).  In the case of SSID with non-alphanumeric characters, I got #1. :bowtie: :clipboard:

**Comment by juliendarrah** [comment]
Did some more investigating last night, want to post it before I forget...

I did a few pass with the wrong SSID and it always show the response page back.
But when the password is wrong the connection seems to get lost because I get a reconnect on it.
see below:

AP Mode Connected
Data Received
Data Received
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Checking Connection Status...
INVALID PASSWORD
AP Mode Reconnected

**Comment by juliendarrah** [comment]
Some time it works....

AP Mode Connected
Data Received
Data Received
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Checking Connection Status...
INVALID PASSWORD
Data Sent
AP Mode Disconnected

**Comment by juliendarrah** [comment]
Even with a valid password, sometime there is a reconnect...

AP Mode Connected
Data Received
Data Received
Connecting...
Connecting...
Connecting...
Connecting...
Connecting...
Checking Connection Status...
CONNECTED!
AP Mode Reconnected

**Comment by juliendarrah** [comment]
The error occurs right after a sent. According to the documentation of the ESP api, when there is an error the recon callback function is called.
I'm not sure to why it happens and how to resend the data.
Need to think about that one more...

**Comment by juliendarrah** [comment]
Did some more test.
I printed out the error returned by the reconnection callback: ESPCONN_TIMEOUT - Timeout.
So I tried a few things without success.
I decided to look for the latest release of the software sdk on Expressif website.
I installed the new release (ESP8266 SDK（esp_iot_sdk_v1.2.0_15_07_03）)
There was something in the release notes that I thought might be related:
Item 5.Revise the scan issue that may cause Wi-Fi connection break.

So I tried it and it does respond faster and also I was not able to get a timeout.
I'm gonna run some more tests before confirming the solution to this issue.
Stay posted...

**Comment by juliendarrah** [comment]
The Release v1.2.0 of the SDK doesn't solve the timeout issue...
But there is a difference in the speed the response is loaded back the the client.
Here remember that the esp module is the server.
Still need to investigate...

**Comment by ishotjr** [comment]
@juliendarrah I set up a guest network on my Airport to allow testing with all-alpha SSID/password; here are the results from the TinyMCU/breadboard ESP:

```
Toggle
Start Configuration AP Mode
AP Mode Connected
AP Mode Connected
Data Received
AP Mode Disconnected
Data Received
AP Mode Disconnected
AP Mode Connected
Data Received
AP Mode Disconnected
CONNECTION TIMED OUT
 ets Jan  8 2013,rst cause:1, boot mode:(3,7)

load 0x40100000, len 30768, room 16
tail 0
chksum 0x5a
load 0x3ffe8000, len 1332, room 8
tail 12
chksum 0x66
ho 0 tail 12 room 4
load 0x3ffe8540, len 7804, room 12
tail 0
chksum 0xd7
csum 0xd7
rl¶
```

**Comment by ishotjr** [comment]
@juliendarrah Here are the results from the ba0sh1/perfboard ESP:

```
Toggle
Start Configuration AP Mode
AP Mode Connected
AP Mode Connected
AP Mode Connected
AP Mode Connected
Data Received
AP Mode Disconnected
Data Received
AP Mode Disconnected
AP Mode Disconnected
AP Mode Disconnected
AP Mode Connected
Data Received
AP Mode Disconnected
CONNECTION TIMED OUT
 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x40100000, len 30768, room 16
tail 0
chksum 0x5a
load 0x3ffe8000, len 1332, room 8
tail 12
chksum 0x66
ho 0 tail 12 room 4
load 0x3ffe8540, len 7804, room 12
tail 0
chksum 0xd7
csum 0xd7
¶¶5¶MEM CHECK FAIL!!!
drl¶
```
